src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -567,130 +514,176 @@ private void SetupActivationInstance(ActivationData result, string grainType, st
-                                    address,
-                                    target,
-                                    activation.GrainInstanceType,
-                                                primary != null ? "Primary Directory partition for this grain is " + primary + ". " : String.Empty,
-                                    address.ToFullString(),
-                                    activation.WaitingCount);
+                                string logMsg =
+                                    string.Format(
+                                        "Tried to create a duplicate activation {0}, but we'll use {1} instead. " +
+                                        "GrainInstanceType is {2}. " +
+                                        "{3}" +
+                                        "Full activation address is {4}. We have {5} messages to forward.",
+                                        address,
+                                        activation.ForwardingAddress,
+                                        activation.GrainInstanceType,
+                                        primary != null
+                                            ? "Primary Directory partition for this grain is " + primary + ". "
+                                            : String.Empty,
+                                        address.ToFullString(),
+                                        activation.WaitingCount);
                                 if (activation.IsUsingGrainDirectory)
                                 {
                                     logger.Info(ErrorCode.Catalog_DuplicateActivation, logMsg);
                                 }
-                                else
-                                {
-                                    if (logger.IsVerbose) logger.Verbose(ErrorCode.Catalog_DuplicateActivation, logMsg);
-                                }
-                                RerouteAllQueuedMessages(activation, target, "Duplicate activation", ex);
-                            }
-                            else
-                            {
-                                logger.Warn(ErrorCode.Runtime_Error_100064,
-                                    String.Format("Failed to RegisterActivationInGrainDirectory for {0}.",
-                                        activation), ex);
-                                // Need to undo the registration we just did earlier
-                                if (activation.IsUsingGrainDirectory)
+                                else if (logger.IsVerbose)
                                 {
-                                    scheduler.RunOrQueueTask(() => directory.UnregisterAsync(address, UnregistrationCause.Force),
-                                        SchedulingContext).Ignore();
+                                    logger.Verbose(ErrorCode.Catalog_DuplicateActivation, logMsg);
```

Comment:
no need to check for IsVerbose, since you are not formatting strings or incurring any penalty by just calling `logger.Verbose`  directly, which already checks for IsVerbose